src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -897,6 +892,128 @@ fn get_var(var_base: &str, host: &str, target: &str) -> Option<OsString> {
         .or_else(|| env::var_os(var_base))
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct OmpOffload {
+    pub target: TargetSelection,
+}
+
+impl Step for OmpOffload {
+    type Output = PathBuf;
+    const IS_HOST: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/llvm-project/offload")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(OmpOffload { target: run.target });
+    }
+
+    /// Compile OpenMP offload runtimes for `target`.
+    #[allow(unused)]
+    fn run(self, builder: &Builder<'_>) -> PathBuf {
+        if builder.config.dry_run() {
+            return PathBuf::from("/");
+        }
+        let target = self.target;
+
+        let LlvmResult { host_llvm_config, .. } = builder.ensure(Llvm { target: self.target });
+
+        let out_dir = builder.llvm_out(target);
+        static STAMP_HASH_MEMO: OnceLock<String> = OnceLock::new();
+        let smart_stamp_hash = STAMP_HASH_MEMO.get_or_init(|| {
+            generate_smart_stamp_hash(
+                builder,
+                &builder.config.src.join("src/llvm-project/offload"),
+                builder.offload_info.sha().unwrap_or_default(),
```

Comment:
I don't think that we need `offload_info` at all, as `offload` is not a submodule, so you can remove `offload_info`. Just use the LLVM SHA, it will be the same as the offload SHA anyway. See [compiler-rt](https://github.com/rust-lang/rust/blob/b1217d1de091e90bb0c187c1f3b196983bac47e6/src/bootstrap/src/core/build_steps/llvm.rs#L1275).